### PR TITLE
Buyer approval endpoints

### PR DESCRIPTION
**Pull request by vraravam**
Added an `approved` status to be marked by an admin. Only once the `approved` flag is set to `true`, can the registered user login into the application. Also added support for filtering the list of users based on these conditions: `persona`, `name`, `login`, `emailId`, `approved`, `city`

### CODE REVIEW

File: src/service/userDetails.js [hunk 1]
```diff
@@ -159,9 +164,33 @@ class UserDetails {
...
+        .catch(error => res.status(400).send(error));
+    } catch (e) {
+      res.status(500).send({ error: e.message });
+    }
+  }
+
   static login(req, res) {
     try {
-      userDetails.findOne({ where: { loginId: req.body.loginId } }).then((user) => {
+      userDetails.findOne({ where: { loginId: req.body.loginId, approved: true } }).then((user) => {
```
Comment (nagarajankannan) [inline src/service/userDetails.js]: I hope we can find the user just by login Id and then decide the code flow based on persona and approved status.

For now the field "approved" describes the buyer's status. So there is a chance that this endpoint might block admin's login if we don't do a data migration setting approved true for admin.

Please correct me, if there is any gape in my understanding.

Comment (vraravam) [reply src/service/userDetails.js]: good catch @nagarajankannan - will make the modification.

Comment (vraravam) [reply src/service/userDetails.js]: Done - please review once again.

**Comment by nagarajankannan** [comment]
**Tasks:**
- [x] Sending SMS after signup and approval 
- [ ] Sending email (Code completed, getting "invalid from" msg91. Have to check whether the transactional email is enabled and the domain is verified)

**Changes proposed:**
- Commented out the userMeta.associate method as it results in error while finding userDetails by primary key. (It looks for a column that doesn't exist)
- Updated default to "defaultValue" in user details schema and migration.

File: src/constants/notification-templates.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (subbu3012) [inline src/constants/notification-templates.js]: Just a suggestion. can we maintain these objects like:
```json
{
  "buyer/seller-signup":{
    "sms": {
      "message": "Lorem message"
    },
    "email": {
      "subject": "Loresm subject",
      "message": "Lorem message"
    }
  }
}
````

**Comment by vraravam** [comment]
> LGTM.

@subbu3012 - I assume you are ok going live without the email functionality? ie only SMS for notifications?

File: src/constants/notification-templates.js (hunk 2 above)
Comment (vraravam) [reply src/constants/notification-templates.js]: Done in last commit.

**Comment by subbu3012** [comment]
> > LGTM.
> 
> @subbu3012 - I assume you are ok going live without the email functionality? ie only SMS for notifications?

> @vraravam - yeah. lets not wait for email to go live. we can fix this at the earliest. also, i guess we are clear on deploying this backend. are we?

**Comment by vraravam** [comment]
not yet - i haven't had a chance to view the video. I was asking more from the "merging this to `master` so that it can be tested with the UI pieces"

**Pull request merged**